File: src/javascript/user/addressbook/delete-address-book.mjs
import { FaxCoreClient, printResult } from '../../common/faxcoreClient.mjs';

if (process.env.FAXCORE_CONFIRM_DESTRUCTIVE !== 'true') {
  throw new Error('Set FAXCORE_CONFIRM_DESTRUCTIVE=true to delete an address book.');
}

const client = new FaxCoreClient();
const result = await client.delete('/api/addressbook', {
  addressBookID: Number(process.env.FAXCORE_ADDRESS_BOOK_ID || 0)
});

printResult('Delete address book', result);

File: src/dotnet/FaxCore.ApiSamples/Samples/User/AddressBook/DeleteAddressBook.cs
using System;
using System.Threading.Tasks;

namespace FaxCore.ApiSamples.Samples.User.AddressBook
{
    internal static class DeleteAddressBook
    {
        public static Task<string> RunAsync(FaxCoreClient client, SampleConfig config)
        {
            RequireDestructiveConfirmation();
            return client.DeleteAsync("/api/addressbook", new { addressBookID = config.AddressBookID });
        }

        private static void RequireDestructiveConfirmation()
        {
            if (!string.Equals(Environment.GetEnvironmentVariable("FAXCORE_CONFIRM_DESTRUCTIVE"), "true", StringComparison.OrdinalIgnoreCase))
            {
                throw new InvalidOperationException("Set FAXCORE_CONFIRM_DESTRUCTIVE=true to delete an address book.");
            }
        }
    }
}
